libsolidity/codegen/YulUtilFunctions.h

```diff
@@ -104,6 +104,21 @@ class YulUtilFunctions
 	std::string overflowCheckedUIntSubFunction();
 
 	std::string arrayLengthFunction(ArrayType const& _type);
+
+	/// @returns the name of a function that resizes an array
+	/// signature: (array, newLen)
+	std::string resizeDynamicArrayFunction(ArrayType const& _type);
+
+	/// @returns the name of a function that will clear the storage area given
+	/// by the start and end parameters
```

Comment:
This prompts a question: What if the distance between start and end is not a multiple of the storage size of the type?

Reply:
hmm good point. Maybe we need to change the parameters to be the actual element count and use the storageBytes / storageSize info to properly clear the data

Reply:
I would change that once I get to different types than uint though